### PR TITLE
40 integrate ai model ros2

### PR DESCRIPTION
**Pull request by Soso0400**
## Description

To integrate the AI model into the car, I developed a python script that load the Yolo v8 model (best.pt of train7).

I create a node to run the python script in the launch file of the jetson.

## Comments

I had errors when doing colcon build because all python dependencies needed was not installed. The dependancies are written in the READ_ME file of the usb_cam

Maybe it will be great to write C++ scripts instead of Python script in the future if we have some problem of performances

## Result images
voiture1
voiture2
voiture3
voiture4
voiture4remontee
voitureSUV
voitureTM

### CODE REVIEW

File: jetsonNano/ros2_ws/src/geicar_start_jetson/launch/geicar.jetson.launch.py [hunk 1]
```diff
@@ -32,10 +32,24 @@ def generate_launch_description():
...
+    
+    # Plate detection node
+    plate_detection_node = Node(
+        package='usb_cam',  # name of the package where you can find the .py script of the plate detection
+        executable='plate_detection.py',  # Name of the Python file
+        name='plate_detection',  # Name of the ROS node
+        output='screen',
+        parameters=[{
+            'camera_topic': '/image_raw',  # If needed you can change the name of the camera topic
+            'detection_threshold': 0.5,
```
Comment (emmacstn) [inline jetsonNano/ros2_ws/src/geicar_start_jetson/launch/geicar.jetson.launch.py:44]: What does the threshold refer to?

**Review by emmacstn — APPROVED** [review]
nice work :)

File: jetsonNano/ros2_ws/src/usb_cam/scripts/plate_detection.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            self.get_logger().error(f"Failed to convert image: {e}")
+            return
+
+        # Applied YOLOv8 on the image
+        results = self.model(frame)
+
+        # Annoted the image with the results
+        annotated_image = frame.copy()
+
+        for result in results:
```
Comment (kekoleko123) [inline jetsonNano/ros2_ws/src/usb_cam/scripts/plate_detection.py:52]: My brain exploded starting here but I trust you 👍

**Review by kekoleko123 — APPROVED** [review]
Very cool !

File: jetsonNano/ros2_ws/src/usb_cam/scripts/plate_detection.py (hunk 2 above)
Comment (Soso0400) [reply jetsonNano/ros2_ws/src/usb_cam/scripts/plate_detection.py:52]: Ahaha the joy of python :)

**Pull request merged**